Storage/DiskImageStore.swift

```diff
@@ -49,92 +53,56 @@ open class DefaultDiskImageStore: DiskImageStore {
-                        continuation.resume(returning: image)
-                    } else {
-                        continuation.resume(throwing: DiskImageStoreErrorCase.invalidImageData(description: "Invalid image data"))
-                    }
-                } catch {
-                    continuation.resume(throwing: error)
-                }
-            }
+        if !self.keys.contains(key) {
+            throw DiskImageStoreErrorCase.notFound(description: "Image key not found")
+        }
+
+        let imagePath = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
+        let data = try Data(contentsOf: imagePath)
+        if let image = UIImage(data: data) {
+            return image
+        } else {
+            throw DiskImageStoreErrorCase.invalidImageData(description: "Invalid image data")
         }
     }
 
-    /// Adds an image for the given key.
-    /// This put is asynchronous; the image is not recorded in the cache until the write completes.
-    /// Does nothing if this key already exists in the store.
     public func saveImageForKey(_ key: String, image: UIImage) async throws {
-        try await withCheckedThrowingContinuation { continuation in
-            queue.async {
-                let size = CGSize(width: image.size.width / 2, height: image.size.height / 2)
-
-                let format = UIGraphicsImageRendererFormat()
-                format.scale = 1
-                let renderer = UIGraphicsImageRenderer(size: size, format: format)
-                let scaledImage = renderer.image { _ in
-                    image.draw(in: CGRect(origin: .zero, size: size))
-                }
-
-                let imageURL = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
-                if let data = scaledImage.jpegData(compressionQuality: self.quality) {
-                    do {
-                        try data.write(to: imageURL, options: .noFileProtection)
-                        self.keys.insert(key)
-                        continuation.resume()
-                    } catch {
-                        continuation.resume(throwing: error)
-                    }
-                } else {
-                    continuation.resume(throwing: DiskImageStoreErrorCase.cannotWrite(description: "Could not write image to file"))
-                }
-            }
+        let size = CGSize(width: image.size.width / 2, height: image.size.height / 2)
+        let format = UIGraphicsImageRendererFormat()
+        format.scale = 1
+        let renderer = UIGraphicsImageRenderer(size: size, format: format)
+        let scaledImage = renderer.image { _ in
+            image.draw(in: CGRect(origin: .zero, size: size))
+        }
+
+        let imageURL = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
+        if let data = scaledImage.jpegData(compressionQuality: self.quality) {
+            try data.write(to: imageURL, options: .noFileProtection)
+            self.keys.insert(key)
```

Comment:
nit; and in fact, there's other `self` usage we can remove in other `func`, so I'll stop commenting on that
```suggestion
        let imageURL = URL(fileURLWithPath: filesDir).appendingPathComponent(key)
        if let data = scaledImage.jpegData(compressionQuality:quality) {
            try data.write(to: imageURL, options: .noFileProtection)
            keys.insert(key)
```